Xamarin.Forms.Platform.iOS/CollectionView/ListSource.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	internal class ListSource : List<object>, IItemsViewSource
+	{
+		public ListSource()
+		{
+		}
+
+		public ListSource(IEnumerable<object> enumerable) : base(enumerable)
+		{
+			
+		}
+
+		public ListSource(IEnumerable enumerable)
+		{
+			foreach (object item in enumerable)
+			{
+				Add(item);
```

Comment:
No way to leverage `AddRange`?

Reply:
Your comment made me curious about the performance of AddRange/InsertRange here; using either one would require doing something like `.Cast<object>` from LINQ, but the internal storage growth might be more efficient. So I whipped up a quick test: https://github.com/hartez/ListSourceTest

From my quick-and-dirty testing, adding the items one-by-one in a `foreach` took about 3/4 the time of AddRange/InsertRange. Feel free to clone that repo and look at my testing; I might have missed something obvious. But it looks like this is the fastest option (which surprises me).

Disclaimer: testing was done on a desktop using StopWatch, so this is pseudo-scientific at best.

Reply:
I ran your project, and I'm getting similar results. Looks like casting `IEnumerable` to `IEnumerable<T>` via LINQ is costly. 

I tried replacing the `foreach` loop with a `MoveNext()/Current` logic, but there was no performance benefit. Also, your last sample is the most efficient partly because the other two run beforehand and the underlying data source does not change. If you comment them out and leave the last one, it will take much longer to finish. There is some initialization overhead which gets gobbled up by the other tests.

Having said that, I think `foreach` here is the right choice.